BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift

```diff
@@ -221,62 +211,61 @@ public final class ConvivaAnalytics: NSObject {
             return
         }
 
-        client.reportError(sessionKey, errorMessage: message, errorSeverity: severity)
+        videoAnalytics.reportPlaybackError(message, errorSeverity: severity)
         if endSession {
             internalEndSession()
         }
     }
 
     /**
-     Puts the session state in a notMonitored state.
+     Puts the session in a notMonitored state.
      */
-    public func pauseTracking() {
-        // AdStart is the preferred way to pause tracking according to conviva.
-        client.adStart(sessionKey,
-                       adStream: .ADSTREAM_SEPARATE,
-                       adPlayer: .ADPLAYER_SEPARATE,
-                       adPosition: .ADPOSITION_PREROLL)
-        client.detachPlayer(sessionKey)
+    public func pauseTracking(isBumper: Bool) {
+        self.isBumper = isBumper
+        let event: String = self.isBumper ?
+        CISConstants.getEventsStringValue(Events.BUMPER_VIDEO_STARTED) :
+        CISConstants.getEventsStringValue(Events.USER_WAIT_STARTED)
+        videoAnalytics.reportPlaybackEvent(event)
         logger.debugLog(message: "Tracking paused.")
     }
 
     /**
-     Puts the session state from a notMonitored state into the last one tracked.
+     Changes the session state from notMonitored state into the previous state.
      */
     public func resumeTracking() {
-        client.attachPlayer(sessionKey, playerStateManager: playerStateManager)
-        // AdEnd is the preferred way to resume tracking according to conviva.
-        client.adEnd(sessionKey)
+        let event: String = self.isBumper ?
+        CISConstants.getEventsStringValue(Events.BUMPER_VIDEO_ENDED) :
+        CISConstants.getEventsStringValue(Events.USER_WAIT_ENDED)
+        videoAnalytics.reportPlaybackEvent(event)
         logger.debugLog(message: "Tracking resumed.")
     }
 
     // MARK: - session handling
     private func setupPlayerStateManager() {
-        playerStateManager = client.getPlayerStateManager()
-        playerStateManager.setPlayerState!(PlayerState.CONVIVA_STOPPED)
-        playerStateManager.setPlayerType!("Bitmovin Player iOS")
-
-        playerStateManager.setCISIClientMeasureInterface?(self)
-
+        videoAnalytics.reportPlaybackMetric(
+            CIS_SSDK_PLAYBACK_METRIC_PLAYER_STATE,
+            value: PlayerState.CONVIVA_STOPPED.rawValue)
+        var playerInfo = [String: Any]()
+        playerInfo[CIS_SSDK_PLAYER_FRAMEWORK_NAME] = "Bitmovin Player iOS"
         if let bitmovinPlayerVersion = playerHelper.version {
```

Comment:
In the meantime, a `sdkVersion` property was added so we always get the correct player version without using the bundle magic: https://cdn.bitmovin.com/player/ios/3/docs/Classes/PlayerFactory.html#/s:14BitmovinPlayer0B7FactoryC10sdkVersionSSvpZ

```suggestion
        if let bitmovinPlayerVersion = PlayerFactory.sdkVersion {
```

Please also remove the relevant code in the `BitmovinPlayerHelper.swift`:
https://github.com/bitmovin/bitmovin-player-ios-analytics-conviva/blob/fc965c92b3b2b68217ac8b301f9ad9016062fb97/BitmovinConvivaAnalytics/Classes/Helper/BitmovinPlayerHelper.swift#L32-L34

Reply:
Ok. We're currently on 3.11.0 in the integration. May I suggest a separate issue for this, as upgrading to 3.28.0 causes a few issues?

Reply:
I see, okay fine. Let's please put the Bitmovin Player upgrade to latest (3.28.0+) in the backlog